File: rust/461.hamming-distance.rs
/*
 * @lc app=leetcode id=461 lang=rust
 *
 * [461] Hamming Distance
 */

// @lc code=start
impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        // Solution 1: using built-in method
        // (x ^ y).count_ones() as i32

        // Solution 2:
        let mut d = 0;
        let mut i = x ^ y;
        while i > 0 {
            d += (i & 1) as i32;
            i >>= 1;
        }
        d
    }
}
// @lc code=end

File: javascript/461.hamming-distance.js
/*
 * @lc app=leetcode id=461 lang=javascript
 *
 * [461] Hamming Distance
 */

// @lc code=start
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function (x, y) {
  let i = x ^ y,
    d = 0;
  while (i) {
    d += i & 1;
    i >>= 1;
  }
  return d;
};
// @lc code=end
